src/inference.jl

```diff
@@ -1986,3 +1408,397 @@ function rxinference(;
+        # if x is a vector, then it simply uses the same value for all elements
+        # However, it is also possible to provide a vector of distributions to set each element individually 
+        x = vague(NormalMeanPrecision),  
+    ),
+)
+
+This argument needs to be a named tuple, i.e. `initmarginals = (a = ..., )`, or dictionary.
+
+- ### `initmessages`
+
+For specific types of inference algorithms, such as loopy belief propagation or expectation propagation, it might be required to initialize (some of) the messages before running the inference procedure in order to break the dependency loop. If this is not done, the inference algorithm will not be executed due to the lack of information and message and/or marginals will not be updated. In order to specify these initial messages, you can use the `initmessages` argument, such as
+```julia
+infer(...
+    initmessages = (
+        # initialize the messages distribution of x as a vague Normal distribution
+        # if x is a vector, then it simply uses the same value for all elements
+        # However, it is also possible to provide a vector of distributions to set each element individually 
+        x = vague(NormalMeanPrecision),  
+    ),
+)
+
+- ### `options`
+
+- `limit_stack_depth`: limits the stack depth for computing messages, helps with `StackOverflowError` for some huge models, but reduces the performance of inference backend. Accepts integer as an argument that specifies the maximum number of recursive depth. Lower is better for stack overflow error, but worse for performance.
+- `pipeline`: changes the default pipeline for each factor node in the graph
+- `global_reactive_scheduler`: changes the scheduler of reactive streams, see Rocket.jl for more info, defaults to no scheduler
+
+- ### `returnvars`
+
+`returnvars` accepts a tuple of symbols and specifies the latent variables of interests. For each symbol in the `returnvars` specification the `rxinference` function will prepare an observable stream (see `Rocket.jl`) of posterior updates. An agent may subscribe on the new posteriors events and perform some actions.
+For example:
+
+```julia
+engine = infer(
+    ...,
+    autoupdates = my_autoupdates,
+    returnvars = (:x, :τ),
+    autostart  = false
+)
+
+- ### `predictvars`
+
+`predictvars` specifies the variables which should be predicted. In the model definition these variables are specified
+as datavars, although they should not be passed inside data argument.
+
+Similar to `returnvars`, `predictvars` accepts a `NamedTuple` or `Dict`. There are two specifications:
+- `KeepLast`: saves the last update for a variable, ignoring any intermediate results during iterations
+- `KeepEach`: saves all updates for a variable for all iterations
+
+Example: 
+
+```julia
+result = infer(
+    ...,
+    predictvars = (
+        o = KeepLast(),
+        τ = KeepEach()
+    )
+)
+```
+
+**Note**: The `predictvars` argument is exclusive for batch setting.
+
+- ### `historyvars`
```

Comment:
`historyvars` is exclusive for streamline setting

Reply:
I think it is said in both short and extended description.